### PR TITLE
Feat/create money primitive obssesion

### PR DESCRIPTION
**Pull request by MarcusViniciusSS**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Resolve the primitive obsession with money.

* **What is the current behavior?** (You can also link to an open issue here)
Use the decimal for everything.

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: src/Services/Catalog/Command/Domain/ValueObjects/Money.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public static bool operator >(Money money, Money other)
+        => money.Value > other.Value;
+
+    public static bool operator <(Money money, Money other)
+        => money.Value < other.Value;
+
+    public static implicit operator string(Money money)
+        => $"{money.Value} {money.Currency.Symbol}";
```
Comment (MarcusViniciusSS) [inline src/Services/Catalog/Command/Domain/ValueObjects/Money.cs:36]: this symbol believe is first then value.

See example below

image

File: src/Services/Order/Command/Domain/ValueObjects/Money.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public static bool operator >(Money money, Money other)
+        => money.Value > other.Value;
+
+    public static bool operator <(Money money, Money other)
+        => money.Value < other.Value;
+
+    public static implicit operator string(Money money)
+        => $"{money.Value} {money.Currency.Symbol}";
```
Comment (MarcusViniciusSS) [inline src/Services/Order/Command/Domain/ValueObjects/Money.cs:36]: the same condition comment previous

File: src/Services/Payment/Command/Domain/ValueObjects/Money.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public static bool operator >(Money money, Money other)
+        => money.Value > other.Value;
+
+    public static bool operator <(Money money, Money other)
+        => money.Value < other.Value;
+
+    public static implicit operator string(Money money)
+        => $"{money.Value} {money.Currency.Symbol}";
```
Comment (MarcusViniciusSS) [inline src/Services/Payment/Command/Domain/ValueObjects/Money.cs:36]: the same condition comment previous

File: src/Services/ShoppingCart/Command/Domain/ValueObjects/Money.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public static bool operator >(Money money, Money other)
+        => money.Value > other.Value;
+
+    public static bool operator <(Money money, Money other)
+        => money.Value < other.Value;
+
+    public static implicit operator string(Money money)
+        => $"{money.Value} {money.Currency.Symbol}";
```
Comment (MarcusViniciusSS) [inline src/Services/ShoppingCart/Command/Domain/ValueObjects/Money.cs:36]: the same condition comment previous

File: src/Services/Payment/Query/Application/UseCases/Events/ProjectPaymentMethodWhenChangedInteractor.cs [hunk 5]
```diff
@@ -0,0 +1 @@
+﻿
```
Comment (MarcusViniciusSS) [inline src/Services/Payment/Query/Application/UseCases/Events/ProjectPaymentMethodWhenChangedInteractor.cs:1]: ??

**Review by AntonioFalcaoJr — APPROVED** [review]

**Pull request merged**